pkg/kubectl/cmd/config/create_authinfo.go

```diff
@@ -124,14 +125,14 @@ func (o createAuthInfoOptions) run() error {
 }
 
 // authInfo builds an AuthInfo object from the options
-func (o *createAuthInfoOptions) modifyAuthInfo(existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
+func (o *createAuthInfoOptions) modifyAuthInfo(cmd *cobra.Command, existingAuthInfo clientcmdapi.AuthInfo) clientcmdapi.AuthInfo {
 	modifiedAuthInfo := existingAuthInfo
 
 	var setToken, setBasic bool
 
-	if o.clientCertificate.Provided() {
-		certPath := o.clientCertificate.Value()
-		if o.embedCertData.Value() {
+	if cmd.Flags().Changed(clientcmd.FlagCertFile) {
```

Comment:
We use that approach heavily to build complex commands that delegate to simpler commands which can also be called individually. I'd hate to see these commands become less reusable and composable.